app/__init__.py

```diff
@@ -197,6 +199,24 @@ def get_spots():
     spot = Spot.query.filter_by(id=beach_id).first()
     spot_data = spot.__dict__
     spot_data.pop('_sa_instance_state', None)
+    ###################################
+    reviews = db.session.query(db.func.count(Review.rating), Review.rating).filter_by(beach_id=beach_id).group_by(Review.rating).all()
+    count = Review.query.filter_by(beach_id=beach_id).count()
+    average = db.session.query(db.func.avg(Review.rating)).filter_by(beach_id=beach_id).all()
+    output = {}
+    for review in reviews:
+      output[str(review[1])] = review[0]
+    output["total"] = count
+    output["average"] = average[0][0]
```

Comment:
like i said in the previous comment, we're probably going to end up removing this, but in general if you know that you are only expecting one result from the db, instead `.all()`, you can use `.first()` or `.one_or_none()`. i'm pretty sure those would also be faster to execute since the db can stop the query once it finds a result. that will prevent you from having to write [0][0].